Source/Core/Common/QoSSession.cpp

```diff
@@ -20,13 +20,13 @@ QoSSession::QoSSession(ENetPeer* peer, int tos_val) : m_peer(peer)
   if (!QOSCreateHandle(&ver, &m_qos_handle))
     return;
 
-  sockaddr_in sin = {};
+  sockaddr_in6 sin6 = {};
 
-  sin.sin_family = AF_INET;
-  sin.sin_port = ENET_HOST_TO_NET_16(peer->host->address.port);
-  sin.sin_addr.s_addr = peer->host->address.host;
+  sin6.sin6_family = AF_INET6;
+  sin6.sin6_port = ENET_HOST_TO_NET_16(peer->host->address.port);
+  memcpy(&sin6.sin6_addr, &peer->host->address.host, sizeof(struct in6_addr));
```

Comment:
Does this still work fine for IPv4?

Reply:
Should work just fine for machines that just have no IPv6 connectivity. That's what IPv4-mapped addresses are for. But same as for the other comment, it will fail if someone deliberately runs with IPv6 hard-disabled in the kernel.